app/src/main/java/com/zeapo/pwdstore/ui/dialogs/PasswordGeneratorDialogFragment.kt

```diff
@@ -4,101 +4,96 @@
-                Toast.makeText(requireActivity(), e.message, Toast.LENGTH_SHORT).show()
-                passwordText.text = ""
+        return MaterialAlertDialogBuilder(requireContext()).run {
+            setTitle(R.string.pwgen_title)
+            setView(view)
+            setPositiveButton(R.string.dialog_ok) { _, _ ->
+                val edit = callingActivity.findViewById<EditText>(R.id.password)
+                edit.setText(passwordText.text)
             }
-            dialog.getButton(AlertDialog.BUTTON_NEGATIVE).setOnClickListener {
-                setPreferences()
-                try {
-                    passwordText.text = generate(callingActivity.applicationContext)[0]
-                } catch (e: PasswordGeneratorExeption) {
-                    Toast.makeText(requireActivity(), e.message, Toast.LENGTH_SHORT).show()
-                    passwordText.text = ""
+            setNeutralButton(R.string.dialog_cancel) { _, _ -> }
+            setNegativeButton(R.string.pwgen_generate, null)
+            create()
+        }.apply {
+            setOnShowListener {
+                generate(passwordText)
+                getButton(AlertDialog.BUTTON_NEGATIVE).setOnClickListener {
+                    generate(passwordText)
                 }
             }
         }
-        return dialog
     }
 
-    private fun setPreferences() {
-        val preferences = ArrayList<String>()
-        if (!(dialog!!.findViewById<CheckBox>(R.id.numerals)).isChecked) {
-            preferences.add("0")
-        }
-        if ((dialog!!.findViewById<CheckBox>(R.id.symbols)).isChecked) {
-            preferences.add("y")
-        }
-        if (!(dialog!!.findViewById<CheckBox>(R.id.uppercase)).isChecked) {
-            preferences.add("A")
-        }
-        if (!(dialog!!.findViewById<CheckBox>(R.id.ambiguous)).isChecked) {
-            preferences.add("B")
-        }
-        if (!(dialog!!.findViewById<CheckBox>(R.id.pronounceable)).isChecked) {
-            preferences.add("s")
-        }
-        if (!(dialog!!.findViewById<CheckBox>(R.id.lowercase)).isChecked) {
-            preferences.add("L")
-        }
-        val editText = dialog!!.findViewById<EditText>(R.id.lengthNumber)
+    private fun generate(passwordField: AppCompatTextView) {
+        setPreferences()
         try {
-            val length = Integer.valueOf(editText.text.toString())
-            setPrefs(requireActivity().applicationContext, preferences, length)
-        } catch (e: NumberFormatException) {
-            setPrefs(requireActivity().applicationContext, preferences)
+            passwordField.text = generate(requireContext().applicationContext)
+        } catch (e: PasswordGeneratorException) {
+            Toast.makeText(requireActivity(), e.message, Toast.LENGTH_SHORT).show()
+            passwordField.text = ""
         }
     }
+
+    private fun isChecked(@IdRes id: Int): Boolean {
+        return dialog!!.findViewById<CheckBox>(id).isChecked
```

Comment:
there should be a `requireDialog()` method from AppCompat that is available here